### PR TITLE
Move vpcPath to NetworkConfigurations CR

### PR DESCRIPTION
**Pull request by seanpang-vmware**
1. Move vpcPath parameter from NetworkInfo CR to NetworkConfiguration CR.
2. Refactor NetworkInfo e2e test to align with new solution.

### CODE REVIEW

**Comment by wenqiq** [comment]
Could you add the TestDone?

**Comment by seanpang-vmware** [comment]
Test Done

> Could you add the TestDone?

updated on commit message

File: build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml [hunk 1]
```diff
@@ -18,3 +18,4 @@ spec:
   vpc: /orgs/default/projects/proj-1/vpcs/vpc-1
   defaultSubnetSize: 32
   vpcConnectivityProfile: /orgs/default/projects/wenqi-test/vpc-connectivity-profiles/default
+  vpcPath: /orgs/default/projects/project-quality/vpcs/19a8a52e-beb0-4396-91ce-5821a15a43db
```
Comment (dantingl) [inline build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml:21]: why add it in spec?

File: build/yaml/crd/vpc/crd.nsx.vmware.com_networkinfos.yaml [hunk 2]
```diff
@@ -61,7 +61,6 @@ spec:
               required:
               - defaultSNATIP
               - name
-              - vpcPath
```
Comment (dantingl) [inline build/yaml/crd/vpc/crd.nsx.vmware.com_networkinfos.yaml:64]: remove vpcPath in next following patch

File: build/yaml/samples/nsx_v1alpha1_networkinfo.yaml [hunk 3]
```diff
@@ -14,5 +14,4 @@ vpcs:
   loadBalancerIPAddresses: 172.26.0.0/26
   name: vpc-d110d5aa-006d-4b59-9caf-424a4fba932c--kube-system
   privateIPs:
-  - 172.26.0.0/16
-  vpcPath: /orgs/default/projects/project-quality/vpcs/19a8a52e-beb0-4396-91ce-5821a15a43db
```
Comment (dantingl) [inline build/yaml/samples/nsx_v1alpha1_networkinfo.yaml:18]: same here

File: pkg/apis/vpc/v1alpha1/networkinfo_types.go [hunk 4]
```diff
@@ -33,14 +33,14 @@ type NetworkInfoList struct {
...
-	// NSX Policy path for VPC.
-	VPCPath string `json:"vpcPath"`
 	// Default SNAT IP for Private Subnets.
 	DefaultSNATIP string `json:"defaultSNATIP"`
 	// LoadBalancerIPAddresses (AVI SE Subnet CIDR or NSX LB SNAT IPs).
 	LoadBalancerIPAddresses string `json:"loadBalancerIPAddresses,omitempty"`
 	// Private CIDRs used for the VPC.
 	PrivateIPs []string `json:"privateIPs,omitempty"`
+	// NSX Policy path for VPC.
+	VPCPath string `json:"vpcPath,omitempty"`
```
Comment (dantingl) [inline pkg/apis/vpc/v1alpha1/networkinfo_types.go:43]: why make it omitempty now?

Comment (seanpang-vmware) [reply pkg/apis/vpc/v1alpha1/networkinfo_types.go:43]: just a safe protection, learnt from wcp develop experience

File: build/yaml/crd/vpc/crd.nsx.vmware.com_networkinfos.yaml (hunk 2 above)
Comment (seanpang-vmware) [reply build/yaml/crd/vpc/crd.nsx.vmware.com_networkinfos.yaml:64]: yes, as we discussed with wenying.

File: build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml (hunk 1 above)
Comment (seanpang-vmware) [reply build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml:21]: fixed

**Review by dantingl — APPROVED** [review]

**Review by timdengyun — APPROVED** [review]

**Pull request merged**